src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpHandler.cs

```diff
@@ -1092,12 +1084,6 @@ private void SaveServerCredentialsToCache(Uri uri, uint authScheme, ICredentials
                 return;
             }
 
-            var cancellationTokenRegistration = state.CancellationToken.Register(s =>
```

Comment:
The current WinHttpHandler is sync pattern.  The cancellation token is checked a few times during the request/response cycle.  But, of course, using sync APIs means that it won't really be cancelled if the request is "in flight" and waiting for a response from the server (at least getting headers). We can't really cancel until a particular WinHTTP API returns back.

Cancellation will be more granular once async pattern is used (#3000).